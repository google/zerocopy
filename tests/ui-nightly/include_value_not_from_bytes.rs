// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#[macro_use]
extern crate zerocopy;

fn main() {}

// Should fail because `UnsafeCell<u32>: !FromBytes`.
const NOT_FROM_BYTES: core::cell::UnsafeCell<u32> =
    include_value!("../../testdata/include_value/data");
