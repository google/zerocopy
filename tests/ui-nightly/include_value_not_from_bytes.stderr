error[E0277]: the trait bound `UnsafeCell<u32>: FromBytes` is not satisfied
  --> tests/ui-nightly/include_value_not_from_bytes.rs:12:5
   |
12 |     include_value!("../../testdata/include_value/data");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     the trait `FromBytes` is not implemented for `UnsafeCell<u32>`
   |     required by a bound introduced by this call
   |
   = help: the following other types implement trait `FromBytes`:
             isize
             i8
             i16
             i32
             i64
             i128
             usize
             u8
           and $N others
note: required by a bound in `AssertIsFromBytes`
  --> tests/ui-nightly/include_value_not_from_bytes.rs:12:5
   |
12 |     include_value!("../../testdata/include_value/data");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `AssertIsFromBytes`
   = note: this error originates in the macro `$crate::transmute` which comes from the expansion of the macro `include_value` (in Nightly builds, run with -Z macro-backtrace for more info)
